pkg/controllers/image/image.go

```diff
@@ -278,6 +287,32 @@ func (h handler) onChangeGitRepo(gitrepo *v1alpha1.GitRepo, status v1alpha1.GitR
 	return status, err
 }
 
+func (h handler) setupKnownHosts(gitrepo *v1alpha1.GitRepo, data []byte) error {
+	tmpdir, err := os.MkdirTemp("", fmt.Sprintf("ssh-%s-%s-", gitrepo.Namespace, gitrepo.Name))
+	if err != nil {
+		return err
+	}
+
+	known := path.Join(tmpdir, "known_hosts")
+	err = os.Setenv("SSH_KNOWN_HOSTS", known)
```

Comment:
The controller is started from a goroutine (I think this code in lasso starts controllers: https://github.com/rancher/lasso/blob/master/pkg/controller/controller.go#L121-L156).

Goroutines share environment variables, as they are the in the same OS process.  I couldn't find any explicit documentation for this, however https://go.dev/play/p/e0L5ixUe4qL illustrates the issue.

By setting the variable `SSH_KNOWN_HOSTS` to a temp file with a random name, we could in theory fail a concurrent second image scan. I think this isn't too unlikely if the interval is low enough and the SSH connection is slow enough.

As a workaround, we could add a lock around this code (which is not good for performance) or pass the known hosts to go-git without the env var somehow?